### ISSUE TITLE
SpherefaceNet-04, SpherefaceNet-06 Release

### ISSUE THREAD

**Issue by zuoqing1988** [wy1iu/sphereface#81]
I have tested SpherefaceNet-04 and SpherefaceNet-06 with and without BatchNorm.

Training on WebFace,
SpherefaceNet-04 WITHOUT BN can easily achive ~98.00% (the paper reports 98.2%),
SpherefaceNet-04 WITH BN        can easily achive ~98.20% (fine-tuning 98.35%),
SpherefaceNet-06 WITHOUT BN can easily achive ~98.50% (the paper doesnot give),
SpherefaceNet-06 WITH BN        can easily achive ~98.70% (fine-tuning 98.80%).

It should be noted that SpherefaceNet-04's forward only costs ~29ms (with BN ~33ms) and SpherefaceNet-06 costs ~49ms (with BN ~55ms), tested on E5-1650V4 3.6GHz, mini-caffe, openblas single thread. 

The attachments are the nets with BN.

[spherefacenet-04-06.zip](https://github.com/wy1iu/sphereface/files/1926231/spherefacenet-04-06.zip)

**Comment by zuoqing1988** [comment]
**A modification from dimension 512 to 256.**
dim256-SpherefaceNet-04 with BN achieves >97.90% (without fine tuning). Forward costs only 9ms~10ms, tested on E5-1650V4 3.6GHz, mini-caffe, openblas single thread. 

[dim256-SpherefaceNet04.zip](https://github.com/wy1iu/sphereface/files/1930189/dim256-SpherefaceNet04.zip)

An important comparsion:
I have implemented my own C++ version of MTCNN, and cropped images for SeetaFace and Sphereface.
The accuracy of SeetaFace is 97.83%, and single forward costs ~110ms.
The accuracy of dim256-SphereFaceNet04 is 97.78% (with flipping 97.93%), single forward costs ~9ms (twice ~19ms)

The attachments are the dim256-SpherefaceNet04

[dim256-SpherefaceNet04.zip](https://github.com/wy1iu/sphereface/files/1930190/dim256-SpherefaceNet04.zip)

**Comment by happynear** [comment]
I've tried the SpherefaceNet-06 with BN yesterday. I modified the fc dimension to 128 and got 98.76%.

**Comment by zuoqing1988** [comment]
@happynear  Thanks.
Currently I am tring to modify every conv and fc laryers to half. It can reduce the forward time to only 25%-30% of the original.

**Comment by silcowitz** [comment]
Are you training these model from scratch/random weights? or using some form of pretraining?

**Comment by wy1iu** [comment]
Thanks for the contribution!

**Comment by twmht** [comment]
@zuoqing1988 

Can you share the training log of SpherefaceNet-06?

**Comment by zuoqing1988** [comment]
[sphereface04_train.zip](https://github.com/wy1iu/sphereface/files/2162196/sphereface04_train.zip)
[sphereface06_train.zip](https://github.com/wy1iu/sphereface/files/2162197/sphereface06_train.zip)
@twmht

**Comment by hrlqq** [comment]
98% +  on which database?

**Comment by lwk98** [comment]
I have trained sphereface06 , but loss no longer converges at 10%. I tried to change the lambda_min to 10, it still has no effect. i don't have a good graphics card (gtx1060), can you give me some advice about how to adjust parameters? Thank you very much! @wy1iu @zuoqing1988

**Comment by zuoqing1988** [comment]
@lwk98 You can find much better models (mxnet or caffe) [here](https://github.com/zuoqing1988/ZQCNN-v0.0)

**Comment by serengil** [comment]
Could those prototxt files be incompatible with the caffemodel shared in the readme file (https://drive.google.com/open?id=0B_geeR2lTMegb2F6dmlmOXhWaVk )?
